src/tesk_core/taskmaster.py

```diff
@@ -4,20 +4,25 @@
 import json
 import os
 import re
+from subprocess import call
 import sys
 import logging
 from kubernetes import client, config
 from tesk_core.job import Job
 from tesk_core.pvc import PVC
 from tesk_core.filer_class import Filer
+from tesk_core.callback_sender import CallbackSender
 
 created_jobs = []
 poll_interval = 5
 task_volume_basename = 'task-volume'
 args = None
 logger = None
+callback = CallbackSender()
```

Comment:
I don't quite understand this code. At this stage, `CallbackSender` misses required information, the URL and the task name. Now, it appears that this is filled in when `newParser()` is called. But when does that happen? And who/what sets the environment variable `CALLBACK_URL` and when? And can you be sure that `data['executors'][0]['metadata']['labels']['taskmaster-name']` is always set and correct when `newParser()` is called? And can you be sure that `callback.url` and `callback.task_id` are always set (and correctly) whenever `callback.send` is called?

Reply:
Also, setting `callback` as a global and changing it in `newParser()` is quite a side effect. Wouldn't it be better to add a setter method to `CallbackSender` and call that somewhere (ideally not in `newParser()` itself, but in the context where `newParser()` is called?